remote-processor/RemoteProcessorServer.cpp

```diff
@@ -89,10 +100,16 @@ void CRemoteProcessorServer::stop()
 
     // Cause exiting of the thread
     uint8_t ucData = 0;
-    write(_aiInbandPipe[1], &ucData, sizeof(ucData));
+    if (not utility::fullWrite(_aiInbandPipe[1], &ucData, sizeof(ucData))) {
+        assert(false);
```

Comment:
What else, pthread_join will most likely never end if the other thread could to be requested to stop itself.